scriptoocloud/src/test/java/edu/asu/diging/scriptoocloud/core/service/impl/DataFileServiceTest.java

```diff
@@ -0,0 +1,110 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import edu.asu.diging.scriptoocloud.core.data.DataFileRepository;
+import edu.asu.diging.scriptoocloud.core.data.DatasetRepository;
+import edu.asu.diging.scriptoocloud.core.exceptions.FileSystemStorageException;
+import edu.asu.diging.scriptoocloud.core.model.impl.DataFile;
+import edu.asu.diging.scriptoocloud.core.model.impl.Dataset;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.support.AnnotationConfigContextLoader;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+
+/**
+ * Test class for DataFileService
+ *
+ * @author John Coronite
+ */
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = AnnotationConfigContextLoader.class)
+class DataFileServiceTest {
+
+    private final Long datasetId = 1L;
+    private final int pageSize = 25;
+    private final int pageNumber = 1;
+
+    @InjectMocks
+    private DataFileService dataFileService;
+
+    @Mock
+    private DataFileRepository dataFileRepository;
+
+    @Mock
+    private DatasetRepository datasetRepository;
+
+    @Mock
+    private FileSystemService fileSystemService;
+
+    @BeforeEach
+    public void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void test_createFile() throws FileSystemStorageException {
+
+        Dataset dataset = Mockito.mock(Dataset.class);
```

Comment:
since dataset is just a domain object and doesn't contain any business logic, I wouldn't mock it. You can just set the properties you need.